### DISCUSSION TITLE
emulating components through GPU

### DISCUSSION THREAD

**Discussion by ghost** [86Box/86Box#1451] [Ideas]
I wonder what would happen if the guest components are emulated by GPU instead of CPU? i guess they'll cost the accuration but might remove slowdowns or will use fps instead.

**Comment by bachimusprime** [comment]
You could only possibly do that with the video card, and even then, it's a bit of a shitfest as to whether or not it'll work as intended :P

**Comment by dgurney** [comment]
To add to what qeeg said, GPUs are optimized for highly parallel workloads, such as machine learning, rendering (duh) and cryptocurrency mining - 86Box isn't one. Additionally, GPUs also have very specialized and limited instruction sets, which adds further difficulty to using them for workloads traditionally reserved for CPUs. 

TL;DR: If there was a hypothetical GPU-only 86Box port, it would be _slower_, not faster.

**Reply by ghost** [reply]
"hypothetical GPU-only" do you mean... GPU-Intensive?

**Reply by dgurney** [reply]
No, not really. 86Box isn't a highly parallel workload, and there is no reasonable way to make it one. As said before, the only component that could be emulated using the GPU is the video card.
